projects/mole/lyf.js

```diff
@@ -238,71 +241,57 @@ async function calLyfTvlSui(api) {
   let poolMap = new Map()
-  poolInfos.forEach(poolInfo => {
-    // const poolId = poolInfo.fields.id.id.replace('0x0', '0x')
-    poolMap.set(poolInfo.fields.id.id, poolInfo)
-  }
+  poolInfos.forEach(poolInfo =>
+    {
+      // const poolId = poolInfo.fields.id.id.replace('0x0', '0x')
+      poolMap.set(poolInfo.fields.id.id, poolInfo)
+    }
   )
 
-  workerInfos.forEach(workerInfo => {
-    let poolId = workerInfo.fields.position_nft.fields.pool
-
-    let coinAamount = 0
-    let coinBamount = 0
-    computeCLMMPositionBalances()
-
-    const [coinA, coinB] = poolMap.get(poolId).type.replace('>', '').split('<')[1].split(', ')
-    api.add(coinA, coinAamount)
-    api.add(coinB, coinBamount)
-
-    // code copied from uni v3 NFT resolver
-    function computeCLMMPositionBalances() {
-      const tickToPrice = (tick) => 1.0001 ** tick
-
+  workerInfos.forEach(workerInfo => 
+    {
       const liquidity = workerInfo.fields.position_nft.fields.liquidity
       const tickLowerIndex = i32BitsToNumber(workerInfo.fields.position_nft.fields.tick_lower_index.fields.bits)
       const tickUpperIndex = i32BitsToNumber(workerInfo.fields.position_nft.fields.tick_upper_index.fields.bits)
-      const bottomTick = tickLowerIndex
-      const topTick = tickUpperIndex
-      const tick = i32BitsToNumber(poolMap.get(poolId).fields.current_tick_index.fields.bits)
-      const sa = tickToPrice(bottomTick / 2)
-      const sb = tickToPrice(topTick / 2)
-
-      if (tick < bottomTick) {
-        coinAamount = liquidity * (sb - sa) / (sa * sb)
-      } else if (tick < topTick) {
-        const price = tickToPrice(tick)
-        const sp = price ** 0.5
-
-        coinAamount = liquidity * (sb - sp) / (sp * sb)
-        coinBamount = liquidity * (sp - sa)
-      } else {
-        coinBamount = liquidity * (sb - sa)
-      }
+      let poolId = workerInfo.fields.position_nft.fields.pool
+      // poolId = poolId.replace('0x0', '0x')
+      const currentSqrtPrice = poolMap.get(poolId).fields.current_sqrt_price
+
+      const coinAmounts = getCoinAmountFromLiquidity(new BN(liquidity), new BN(currentSqrtPrice), tickIndexToSqrtPriceX64(tickLowerIndex), 
+          tickIndexToSqrtPriceX64(tickUpperIndex))
+
+      let coinAamount = coinAmounts.coinA
+      let coinBamount = coinAmounts.coinB
+      
+      const [coinA, coinB] = poolMap.get(poolId).type.replace('>', '').split('<')[1].split(', ')
+      api.add(coinA, coinAamount.toString())
+      api.add(coinB, coinBamount.toString())
     }
-  }
   )
 
   // calculate the Vault TVL.
 
   const vaultInfoIds = addresses.Vaults.map(valut => valut.vaultInfo)
   const vaultInfos = await sui.getObjects(vaultInfoIds)
-
+  
   for (let i = 0; i < vaultInfos.length; i++) {
     const baseToken = addresses.Vaults[i].baseToken
     const tokenAmount = vaultInfos[i].fields.value.fields.coin
-    api.add(baseToken, tokenAmount)
+    const vaultDebtVal  = vaultInfos[i].fields.value.fields.vault_debt_val
```

Comment:
what does this value represent?

Reply:
Hey @g1nt0ki , thanks for the review. This value represent the borrowed part in Mole savings vaults. It is a mistake we didn't add it up before. It is a fixing for vaults value. Thank you.

Reply:
but the borrowed tokens dont leave the platform but are used as LP and you are already adding LP value right?

Reply:
Hi @g1nt0ki , thank you for your review.

1) For the vault_debt_val, sorry that it may cause some misunderstanding. It is mToken value staked in Mole fairlauch. It was missing before. It is actually should be taken into calculated in the Mole savings vaults vaule. 

2) For the Aptos chain TVL, if it should be kept for history tracking purpose, I'll submit a code fix later that keep Mole in Aptos TVL.